src/encryption.rs

```diff
@@ -1,53 +1,72 @@
+use crate::ConfigureError;
 use log::debug;
 use sodiumoxide::base64::Variant;
 use sodiumoxide::base64::{decode, encode};
 use sodiumoxide::crypto::secretbox;
+use std::fmt;
 use std::fs::{read, write};
-use std::io::{Error, ErrorKind};
 use std::path::Path;
 
 pub fn init() {
     sodiumoxide::init().expect("Unable to initialize libsodium");
 }
 
-pub fn generate_key() -> String {
+pub fn generate_key() -> EncryptionKey {
```

Comment:
It'd be great if we return a `Result` here with a proper error instead of utilizing the `expect` unless you think it's very unlikely for `secretbox::gen_key()` call to fail, which sounds reasonable, in which case maybe we should communicate that either in `expect` or as a comment.

Reply:
It should be _very_ rare, so I'm ok to allow a crash if it fails. Added a comment explaining this in 89e6786817e687bbb354d0ef87d33d39714bfc06.